### DISCUSSION TITLE
Azure active directory support

### DISCUSSION THREAD

**Discussion by jakubkoje** [adonisjs/core#2276] [Help] [answered]
Hello, I want to secure my endpoints with azure ad. I saw that Microsoft has a package for passportjs. AFAIK that package is not usable in Adonis. Is there a alternative for it? Thanks.

**Comment by anthrogan** [comment, marked as answer]
I suggest using Microsoft's MSAL library in your front end. It's a straight forward JWT token type auth and is baked in to lots of features of MS, like refresh tokens and querying MS Graph etc.

Verifying the token in the backend is straight forward too, it can be done a few ways. You can use the recommended way from MS, which is verifying the `kid` key which can be fiddly querying their api for the valid keys as they change every 24hours, but its the most secure. Or you could just decrypt the token and check if the user has a company email address for example. This can all be done using middleware's.

**Reply by jakubkoje** [reply]
https://www.voitanos.io/blog/validating-azure-ad-generated-oauth-tokens/
Is this the recommended way you are talking about?

**Reply by anthrogan** [reply]
Yep
